File: java/ConverteDiaSemana.java
/**
 * Programa Java retorna o dia da semana em extenso. Domingo igual 1 e Sábado 7.
 *
 * Objetivos:
 *   - uso de variáveis e switch (múltiplas condições);
 *
 * Módulo: Lógica de programação | Capítulo: XXX | Página: XXX
 * Author: Eder Magalhães
 */
public class ConverteDiaSemana {

    public static void main(String[] args) {
        //modifique essa variável para testar o programa
        int diaSemana = 2;
        String dia = "";

        //armazena o dia
        switch(diaSemana) {
	    case 1: dia = "Domingo"; break;
	    case 2: dia = "Segunda-feira"; break;
	    case 3: dia = "Terça-feira"; break;
	    case 4: dia = "Quarta-feira"; break;
	    case 5: dia = "Quinta-feira"; break;
	    case 6: dia = "Sexta-feira"; break;
	    case 7: dia = "Sabado"; break;
	    default: dia = "Dia invalido!";
        }

        //armazena se é dia útil ou fim de semana
        switch(diaSemana) {
	    case 2:
	    case 3:
	    case 4:
	    case 5:
	    case 6: dia += " (dia útil)"; break;
	    case 1:
	    case 7: dia += " (fim de semana)"; break;
        }
        System.out.println(dia);
    }
    
}

File: python/converteDiaSemana.py
# -*- coding: utf-8 -*-

# Programa Python retorna o dia da semana em extenso. Domingo igual 1 e Sábado 7.
# Python não possui switch, utilizo if c/ elif no lugar.
# Objetivos:
#   - uso de multiplas condições boolean;
#
# Acesse e execute esse código Python em: http://pythonfiddle.com/converte-dia-semana-ap/
#
# Módulo: Lógica de programação
# Author: Eder Magalhães

# modifique essa variável para testar o programa
diaSemana = 2;
dia = "";

if diaSemana == 1:
    dia = "Domingo";
elif diaSemana == 2:
    dia = "Segunda-feira";
elif diaSemana == 3:
    dia = "Terça-feira";
elif diaSemana == 4:
   dia = "Quinta-feira";
elif diaSemana == 5:
   dia = "Quarta-feira";
elif diaSemana == 6:
   dia = "Sexta-feira";
elif diaSemana == 7:
   dia = "Sabado";
else:
   dia = "Dia invalido!";

# complementa, armazena se é dia útil ou fim de semana
if diaSemana in (2, 3, 4, 5):
   dia += " (dia útil)"
elif diaSemana in (1,7):
   dia += " (fim de semana)"

print dia

